instrumentedProfilingLibrary/Time Profiler/ProfileDataManager.cpp

```diff
@@ -0,0 +1,71 @@
+#include "ProfileDataManager.h"
+#include <iostream>
+
+using namespace std;
+
+ProfileDataManager* ProfileDataManager::profiledatamanager = nullptr;
+
+ProfileDataManager::ProfileDataManager(const string name){
+	
+	myfile.open("time.txt");
+	
+	if (myfile.is_open()) {
+	    cout << "myfile is open \n" << endl;
+	}
+
+}
+
+
+ProfileDataManager:: ~ProfileDataManager() {

```

Comment:
Interesting, but how will this destructor be called? So far, it seems to not be called, hence you'll never actually close the file as intended.

Reply:
I had written this with the assumption in mind that usually the  'destructor is called automatically' when the object goes out of scope. I double checked with `cout` commands to see if it is called and indeed it is not called. Could you help me figure out why this is not the case?
To deal with file closing, I have moved it to the main cpp file instead of placing it here in the destructor.